### PR TITLE
Make ifupdown optional for udev users

### PR DESCRIPTION
**Pull request by mpatnode**
So it seems that ifupdown isn't compatible with udev which I was reluctant to remove on my ReadyNAS OS.   Not sure what huge value you get vs ifconfig (maybe the --force? option) but it was easy enough to make it optional.

### CODE REVIEW

File: network_check.sh [hunk 1]
```diff
@@ -15,7 +17,7 @@ clear
 ###
 
 # Set gateway_ip to the gateway that you want to check to declare network working or not
-gateway_ip='www.google.com'
+gateway_ip='1.1.1.1'
```
Comment (mpatnode) [inline network_check.sh:20]: IMHO:  No need to restart things because of a DNS failure.

Comment (ltpitt) [reply network_check.sh:20]: Thanks for the contribution, @mpatnode!

All very good points and, to be honest...

I think we should get rid of all the requirements possible and make the code as universally working as possible.
What do you think?

**Review by ltpitt — CHANGES_REQUESTED** [review]
Thanks for the contribution, @mpatnode!

All very good points and, to be honest...

I think this is a very good direction and that we should get rid of all the requirements possible and make the code as universally working as possible.
Therefore I'd suggest completely removing ifupdown in order to simplify code and its future maintenance and increase its compatibility.

What do you think?

**Comment by mpatnode** [comment]
Fine by me.  I don't have a Raspberry Pi to test with, but I assume the OS came with standard ifconfig?   Change coming.

**Comment by ltpitt** [comment]
> Fine by me. I don't have a Raspberry Pi to test with, but I assume the OS came with standard ifconfig? Change coming.

Mmmmh seems like this could be a problem :/

https://raspberrypi.stackexchange.com/questions/1409/easiest-way-to-show-my-ip-address

**Comment by mpatnode** [comment]
So it looks like this is a can of worms.   ifconfig has been deprecated for "ip", so maybe the way it is now is for the best.

**Comment by mpatnode** [comment]
Try:
ip link set wlan0 up
ip link set wlan0 down

**Comment by mpatnode** [comment]
This might do the trick:
 ip link set wlan0 down
 ip link set wlan0 up

**Comment by ltpitt** [comment]
> So it looks like this is a can of worms. ifconfig has been deprecated for "ip", so maybe the way it is now is for the best.
Agreed but we should also not leave other users in the cold.

I think that the pre-requisite check section of the script should find out which tools are available on the pc running it and function correctly without user having to configure / change parts of it.
I will install the newest raspbian on a raspberry pi and try all the old and new commands.

In the meantime we can complete this merge since it contains many other nice features.
For the moment being your idea about using a flag to switch between UDEV / ifupdown seems to be the cleanest solution that keeps other users functionality intact.

**Pull request merged**